addons/token-exchange/s3_secret_handler.go

```diff
@@ -0,0 +1,132 @@
+package addons
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"strings"
+
+	routev1 "github.com/openshift/api/route/v1"
+	"github.com/red-hat-storage/odf-multicluster-orchestrator/controllers/common"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/klog/v2"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+type s3SecretHandler struct {
+	spokeClient client.Client
+}
+
+const (
+	S3SecretHandlerName            = "s3"
+	S3DefaultBlueSecretMatchString = "odrbucket"
+	ObjectBucketClaimKind          = "ObjectBucketClaim"
+	S3BucketName                   = "BUCKET_NAME"
+	S3BucketRegion                 = "BUCKET_REGION"
+	S3RouteName                    = "s3"
+	DefaultS3EndpointProtocol      = "https"
+	// TODO use storage cluster name and namespace from mirror peer: https://github.com/red-hat-storage/odf-multicluster-orchestrator/pull/55
+	StorageClusterName      = "ocs-storagecluster"
+	StorageClusterNamespace = "openshift-storage"
```

Comment:
Rebase on top of that PR and update this.

Reply:
ack